### ISSUE TITLE
AI Image Recognition for Vehicle Number Plate Verification

### ISSUE THREAD

**Issue by Taherabharmal** [egovernments/Digit-Core#114]
**Description**
The purpose of this requirement document is to outline the specifications and acceptance criteria for building an AI image recognition feature for no. plate of sanitation vehicles on the DIGIT Sanitation platform. The feature aims to enhance the verification process of vehicle number plates entered manually by introducing a photo upload functionality for verification purposes. The AI image recognition algorithm will capture the detected number from the uploaded image, ensuring accurate and reliable verification. 

Actors in DIGIT Sanitation FSM:-
1. Citizen: Requests septic tank desludging services and receives notifications related to waste management.
2. Sanitation Worker: Responsible for performing waste management tasks safely and efficiently.
3. ULB Employee: Manages service requests and maintains citizen information for efficient service delivery.
4. Waste Management Vendors/Desludging Operators: Assign tools and services for effective waste management.
5. Feacal Sludge/Co-treatment Plant/Centre Operator: Oversees waste treatment, adherence to schedules, and treatment quality.
6. Administrators: Govern the waste management value chain and monitor compliance.

**Goals**
[DIGIT FSM](https://sanitation.digit.org/products) is an open-source platform designed to digitize waste management operations. The platform facilitates coordination among stakeholders, ensuring transparency and accountability in the waste management value chain. The proposed feature focuses on improving the verification process for vehicles arriving at the Fecal Sludge Treatment Plant (FSTP), particularly against those coming in without corresponding requests. 

**Actors:**
1. Image Recognition Accuracy:
   a. The AI image recognition algorithm should achieve a minimum accuracy rate of 95% in detecting the vehicle number from the uploaded photo.
2. Integration and Data Handling:
   a. The image recognition feature should seamlessly integrate with the existing FSM platform. 
   b. Uploaded images should be securely processed and stored, following best practices for data privacy and protection.
3. Performance and Response Time:
   a. The feature should deliver prompt results, providing real-time verification feedback to users. 
   b. The system should handle a significant volume of image recognition requests efficiently.
4. Error Handling:
   a. Proper error messages should be displayed if the image recognition process fails or encounters any issues.
   b. Users should be able to retry the verification process if necessary.

**Expected Outcome**
To address the problem, we propose implementing an **AI image recognition feature** for vehicle number plate verification. The feature will require users to upload a photo of the vehicle number plate. The system will detect the vehicle number from the uploaded image. The user can edit this if there is a gap.

**Acceptance Criteria**

1. The AI image recognition algorithm achieves an accuracy rate of **at least 95%** in detecting vehicle numbers from the uploaded images.
2. The feature integrates seamlessly with the existing FSM platform, ensuring smooth workflow and data handling.
3. The verification process provides real-time results, promptly notifying users of the verification status.

**Implementation Details**
Currently, vehicles arriving at the FSTP without corresponding applications in the system can enter dummy vehicle numbers to receive payment. There is no robust verification process to ensure the accuracy or existence of the entered vehicle numbers.

**Mockups / Wireframes**
[Include links to any visual aids, mockups, wireframes, or diagrams that help illustrate what the final product should look like. This is not always necessary, but can be very helpful in many cases.]
[Please note that the below section of the ticket has to be in the format as mentioned as it is key to enabling proper listing of the project. Please only choose the options mentioned under the headings wherever applicable.]

**Product Name**
DIGIT Sanitation

**Project Name**
AI Image Recognition for Vehicle Number Plate Verification

**Organization Name:**
eGov Foundation

**Domain**
DIGIT Sanitation

**Tech Skills Needed:**
Python Libraries

**Mentor(s)**
Tahera

**Complexity**
[Medium]

**Category**
[CI/CD], [Integrations], [Performance Improvement], [Security], [UI/UX/Design], [Bug], [Feature], [Documentation], [Deployment], [Test], [PoC]

Sub Category
Pick one or more of [API], [Database], [Analytics], [Refactoring], [Data Science], [Machine Learning], [Accessibility], [Internationalization], [Localization], [Frontend], [Backend], [Mobile], [SEO], [Configuration], [Deprecation], [Breaking Change], [Maintenance], [Support], [Question], [Technical Debt], [Beginner friendly], [Research], [Reproducible], [Needs Reproduction].

https://sanitation.digit.org/products/faecal-sludge-management-fsm/fsm-user-manual/septage-treatment-plant-operator-user-manual
https://sanitation.digit.org/
https://sanitation.digit.org/platform
https://sanitation.digit.org/platform/architecture
https://sanitation.digit.org/products/faecal-sludge-management-fsm/fsm-core-service-configuration/fsm-dss-technical-documentation

**Comment by vkivaturi** [comment]
Will the image of vehicle number plate get taken through a specific mobile app? If yes, we can include geo tagging and timestamp the photo so that it can be confirmed that the vehicle indeed made it to the pickup point and time at which it reach various stages of the process.

**Comment by Taherabharmal** [comment]
yes, this will be done through the FSM app, which is currently mobile-responsive. This is an interesting suggestion - we are in the process of building vehicle tracking as well.

**Comment by Taherabharmal** [comment]
There are actually 2 actors in this - 1 is the person who is providing de-sludging services i.e the transporter, and the second is the plant operator who is confirming that vehicles entered the plant and disposed at the right place. 

The plant operator usually enters the vehicle number right now for vehicles to confirm vehicles came in. This is for vehicles that might be unregistered and not a part of the system. For this, we have observed that plant operators are entering numbers in random and we need improved verification for the same.

For vehicles that are registered and using the app, the point you suggested in useful where if a photograph is taken with geotag and time stamp, then intervention by hte plant operator to confirm disposal can be avoided.

**Comment by KDwevedi** [comment]
comment

**Comment by masterismail** [comment]
hi @Taherabharmal is this still available to take up ?
